src/object/obj_internal.h

```diff
@@ -412,11 +417,16 @@ obj_ptr2hdl(struct dc_object *obj)
 	return oh;
 }
 
-static inline daos_epoch_t
-dc_io_epoch(void)
+static inline void
+dc_io_epoch_set(struct dc_obj_epoch *epoch)
 {
-	return (srv_io_mode != DIM_CLIENT_DISPATCH) ?
-			DAOS_EPOCH_MAX : crt_hlc_get();
+	if (srv_io_mode == DIM_CLIENT_DISPATCH) {
+		epoch->oe_value = crt_hlc_get();
+		epoch->oe_uncertain = true;
```

Comment:
DIM_CLIENT_DISPATCH is only used for old IO mode without guarantee replicas consistency. So setting "oe_uncertain" as true for that seems unnecessary.

Reply:
OK, will set it to false and add a comment.